### PR TITLE
Speedup update_codecgen by precomputing dependencies.

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #25940

This speeds up update_codecgen script by ~20-25%. (1m-1m30s).

It's not much but if we multiply it by number of runs it starts getting bigger...

@lavalamp @thockin - can one of you please take a look?

### CODE REVIEW

File: hack/update-codecgen.sh [hunk 1]
```diff
@@ -61,20 +61,22 @@ number=${#generated_files[@]}
...
   visited[${i}]=false
+  file="${generated_files[${i}]/\.generated\.go/.go}"
+  deps[${i}]=$(go list -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}' ${file} | grep "^${my_prefix}")
 done
 ###echo "DBG: found $number generated files"
 ###for f in $(echo "${generated_files[@]}" | sort); do
 ###    echo "DBG:   $f"
 ###done
 
+# Precompute dependencies for all directories.
```
Comment (mikedanese) [inline hack/update-codecgen.sh]: This comment seems like it should be moved up a few lines.

**Comment by mikedanese** [comment]
lgtm beside comment nit

**Comment by wojtek-t** [comment]
@mikedanese - comment applied PTAL

**Comment by mikedanese** [comment]
This is attached to a p0 issue so labeling p0

**Pull request merged**